mypy/suggestions.py

```diff
@@ -136,6 +137,47 @@ def get_return_types(typemap: Dict[Expression, Type], func: FuncDef) -> List[Typ
     return finder.return_types
 
 
+class ArgUseFinder(TraverserVisitor):
+    """Find all the types of arguments that each arg is passed to.
+
+    This is extremely simple minded but might be effective anyways.
+    """
+    def __init__(self, func: FuncDef, typemap: Dict[Expression, Type]) -> None:
+        self.typemap = typemap
+        self.arg_types = {
+            arg.variable: [] for arg in func.arguments
+        }  # type: Dict[SymbolNode, List[Type]]
+
+    def visit_func_def(self, o: FuncDef) -> None:
+        # Skip nested functions
+        pass
```

Comment:
Do we really need to skip nested functions here? I could imagine something like:
```python
def g(x: int) -> int: ...

def f(x):
    def nested():
        g(x)
```
Since you are matching my the `Var` node, this seems safe.

Reply:
Yeah, good point.